lib/SparkPost/SparkPost.php

```diff
@@ -64,11 +65,16 @@ public function setupUnwrapped ($endpoint) {
    * Merges passed in headers with default headers for http requests
    */
   public function getHttpHeaders() {
+
     $defaultOptions = [
         'Authorization' => $this->config['key'],
         'Content-Type' => 'application/json',
     ];
 
+    if( $this->config['subaccount_id'] ){
```

Comment:
There are certainly some cases where it is ignored however its not always the case where you will need to change API keys.  Depending on the API master accounts are able to set things on behalf of a subaccount using the master account api key and a subaccount id.  Thats not how the transmissions api currently works but we do want to expand and include other apis.  

If you provide a subaccount api key, it doesn't matter if you set the subaccount header or not, the subaccount for the api key is what is used.